Remove docs about legacy tools

felicitymay opened the pull request:
This pull request removes information about unsupported tools from the docs. For details, see the linked issue.

mchammer01 reviewed (approved):
@felicitymay - this LGTM ✨ 

I found a few more instances of `legacy` and I think that we should be ok to leave them (as they may apply to LGTM.com and LGTM Enterprise), but as I'm not sure, I've decided to mention them here:

- `docs\codeql\ql-language-reference\name-resolution.rst` - line 78

- `docs\codeql\codeql-cli\getting-started-with-the-codeql-cli.rst` - lines 210 and 213

- `docs\codeql\ql-language-reference\ql-language-specification.rst` - paragraph starting on line 99

- `docs\codeql\reusables\index-files-note.rst` - lines 2 and 6

felicitymay commented:
Thanks for the review @mchammer01 - good points. I hadn't thought to look for "legacy" ✨ 

Looking at the information about `queries.xml` here, https://github.com/github/codeql/blob/main/docs/codeql/ql-language-reference/ql-language-specification.rst#library-path, it sounds as if this file is still relevant for LGTM. I think the only extra change that we might want to make is to remove the references to ODASA in https://github.com/github/codeql/blob/main/docs/codeql/ql-language-reference/ql-language-specification.rst#library-path.

I'll check with @github/codeql-core and @AlonaHlobina about this.

AlonaHlobina commented:
Thank you for doing this! 
Unfortunately, I am not fully aware of the ODASA deprecation plan and what part of it LGTM Ent may still use. @adityasharad, don't you mind taking a look?

adityasharad commented on docs/codeql/ql-language-reference/ql-language-specification.rst:
```diff
@@ -71,23 +71,14 @@ language="java"/>``).
 will not be recognized when using legacy tools that are not based
-on the CodeQL CLI (that is, LGTM.com, LGTM Enterprise, ODASA, CodeQL for
-Eclipse, and CodeQL for Visual Studio).
+on the CodeQL CLI (that is, LGTM.com and LGTM Enterprise).
```
LGTM (both forms) now uses CodeQL.
```suggestion
on the CodeQL CLI.
```

hmakholm replied:
Perhaps we should just get rid of the entire sentences -- I don't think we have any active products where the QL engine is _not_ CLI based these days. (Except for tail-end support of old LGTM-e releases, but the language specification doesn't completely apply to them anyway).

felicitymay replied:
As suggested, I've removed this sentence.

adityasharad commented on docs/codeql/support/notes-ql-users.rst:
```diff
@@ -1,91 +0,0 @@
-Notes for legacy QL CLI users
```
This was useful for a few straggling customers, but I think they've all moved off the legacy CLI now.

adityasharad reviewed (dismissed):
Looks good. One minor change: LGTM doesn't use the legacy CLI.

felicitymay commented:
I think that I've addressed all feedback and that this is ready to merge, but it will need an approving review.

I've updated the preview (links in the internal issue).
